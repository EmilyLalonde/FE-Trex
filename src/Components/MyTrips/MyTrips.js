import React, { Component } from 'react';
import { 
  StyleSheet, 
  Text, 
  View, 
  ScrollView,
  TouchableOpacity 
} from 'react-native';
import Footer from '../Footer/Footer';
import Header from '../Header/Header';

export default class MyTrips extends Component {

  constructor(props) {
    super(props)
    this.tripsElements = [],
    this.tripsList = [
      {name: 'South America'},
      {name: 'Madagascar'},
      {name: 'Southeast Asia'}
    ]
  }

  generateTripsElements = (list) => {
    return list.map(trip => {
      return (
      <TouchableOpacity style={styles.tripButton}>
        <Text style={styles.text} key={trip.name}>{trip.name}</Text>
      </TouchableOpacity>
      )
    })
  }

  componentDidMount = () => {
  }
  
  render() {
    const {navigate} = this.props.navigation;
    
    this.tripsElements = this.generateTripsElements(this.tripsList)

    return(
      
      <View style={styles.container}>

        <Header />

        <ScrollView>
          <Text style={styles.title}>My Trips</Text>
          <View>
            {this.tripsElements.length > 0 && this.tripsElements}
          </View>
          <TouchableOpacity style={styles.addTripButton}>
            <Text style={styles.text}>Add a New Trip</Text>
          </TouchableOpacity>
        </ScrollView>
  
        <Footer navigate={navigate} />
        
    </View>

    )
  }
}

const styles = StyleSheet.create({
  title: {
    fontSize: 30
  }, 
  container: {
    flex: 1,
    backgroundColor: '#000000',
    alignItems: 'stretch',
    justifyContent: 'flex-start'
    
  }, 
  text: {
    color: 'white',
    textAlign: 'center',
    paddingVertical: 10,
    fontSize: 20
  },
  title: {
    textAlign: 'center',
    fontSize: 30,
    color: 'white',
    paddingVertical: 10
  },
  tripButton: {
    borderWidth: 1,
    borderColor: 'white',
    borderRadius: 8,
    marginVertical: 10
  },
  addTripButton: {
    backgroundColor: '#1C4263',
    borderWidth: 1,
    borderColor: 'white', 
    color: 'white',
    borderRadius: 8
  }
})
