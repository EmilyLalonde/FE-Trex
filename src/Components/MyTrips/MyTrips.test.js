import React from 'react';
import 'react-native';
import { MyTrips } from './MyTrips';
import { shallow } from 'enzyme';

describe('MyTrips', () => {
  it('should return true', () => {
    expect(true).toEqual(true)
  })

  it('should match the snapshot when loading', () => {
    let wrapper = shallow(<MyTrips navigation={{ navigate: jest.fn() }} />)

    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when trips array is populated', () => {
    let wrapper = shallow(<MyTrips navigation={{ navigate: jest.fn() }} />)

    wrapper.instance().setState({ trips: [{id: 1, name: 'Paris'}, {id: 2, name: 'Rome'}]})

    expect(wrapper).toMatchSnapshot();
  })


  describe('componentDidMount', () => {
    it('should fire fetchMyTrips with the user id', () => {
      
    })
  })
})