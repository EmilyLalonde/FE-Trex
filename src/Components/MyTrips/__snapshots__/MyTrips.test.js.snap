// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTrips should match the snapshot when loading 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <WandererHeader />
  <ScrollViewMock>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      My Trips
    </Text>
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        Loading ...
      </Text>
      <Image
        alt="Loading..."
        source={
          Object {
            "testUri": "../../../assets/wanderer_spinner.gif",
          }
        }
      />
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "backgroundColor": "#1C4263",
          "borderColor": "white",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "white",
          "marginBottom": 10,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        Add a New Trip
      </Text>
    </TouchableOpacity>
  </ScrollViewMock>
  <WandererFooter
    navigate={[MockFunction]}
  />
</View>
`;

exports[`MyTrips should match the snapshot when trips array is populated 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <WandererHeader />
  <ScrollViewMock>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      My Trips
    </Text>
    <View>
      <TouchableOpacity
        activeOpacity={0.2}
        key="0Paris"
        style={
          Object {
            "borderColor": "white",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginVertical": 10,
          }
        }
      >
        <Text
          key="Paris"
          onPress={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "paddingVertical": 10,
              "textAlign": "center",
            }
          }
        >
          Paris
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="1Rome"
        style={
          Object {
            "borderColor": "white",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginVertical": 10,
          }
        }
      >
        <Text
          key="Rome"
          onPress={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "paddingVertical": 10,
              "textAlign": "center",
            }
          }
        >
          Rome
        </Text>
      </TouchableOpacity>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "backgroundColor": "#1C4263",
          "borderColor": "white",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "white",
          "marginBottom": 10,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        Add a New Trip
      </Text>
    </TouchableOpacity>
  </ScrollViewMock>
  <WandererFooter
    navigate={[MockFunction]}
  />
</View>
`;
